Sources/ArcGISToolkit/Components/Authentication/CertificatePickerViewModifier.swift

```diff
@@ -92,11 +103,40 @@ import ArcGIS
         challenge.resume(with: .cancel)
     }
 
-    private func showCertificateImportError(_ error: Error?) async {
+    private func showCertificateError(_ error: CertificateError) async {
         // This is required to prevent an "already presenting" error.
         try? await Task.sleep(nanoseconds: 100_000)
-        certificateImportError = error as? CertificateImportError
-        showCertificateImportError = true
+        certificateError = error
+        showCertificateError = true
+    }
+}
+
+extension CertificateImportError: LocalizedError {
+    public var errorDescription: String? {
+        switch self {
+        case .invalidData:
+            return String(localized: "The certificate file was invalid.", bundle: .module)
+        case .invalidPassword:
+            return String(localized: "The password was invalid.", bundle: .module)
+        default:
+            return SecCopyErrorMessageString(rawValue, nil) as String? ?? String(
+                localized: "The certificate file or password was invalid.",
+                bundle: .module
+            )
+        }
+    }
+}
+
+extension CertificatePickerViewModel.CertificateError: LocalizedError {
+    public var errorDescription: String? {
```

Comment:
```suggestion
    var errorDescription: String? {
```

Reply:
`public` has to be there because it is a requirement of `LocalizedError`.

Reply:
That's not correct. The declaration needs to match the access level of the type, which is internal, so the declaration itself can be internal.

Reply:
you are correct. It was giving error for `CertificateImportError`'s `errorDescription`.